tests/installation/ipxe_install.pm

```diff
@@ -47,12 +47,12 @@ sub poweron_host {
 
 sub set_pxe_boot {
     while (1) {
-        my $stdout = ipmitool('chassis bootparam get 5');
-        last if $stdout =~ m/Boot Flag Valid.*Force PXE/s;
         diag "setting boot device to pxe";
         my $options = get_var('IPXE_UEFI') ? 'options=efiboot' : '';
         ipmitool("chassis bootdev pxe ${options}");
         sleep(3);
+        my $stdout = ipmitool('chassis bootparam get 5');
+        last if $stdout =~ m/Force PXE/s;
```

Comment:
Thanks for the review!
The reason for the change is that: on *some* machines, when the SUT is set next boot by efibootmgr, eg to a usb device, the information will NOT be reflected into `ipmitool('chassis bootparam get 5')`, so the ipmitool bootparam output is not up-to-date. But if after efibootmgr controls the boot order, and then use ipmitool to set boot order again, the `ipmitool('chassis bootparam get 5')` is up to date and can control boot order as it expects.
I test on different machines, and the behavior is not exactly the same, but at least above workaround works well. Since we have many different machines from different vendor and with different spec, I expect that this change can make us more guaranteed to boot into pxe, although may sacrifice one time of `ipmitool("chassis bootdev pxe ${options}");`, but worthwhile I think.

Reply:
Thanks for the review!
The reason for the change is that: on *some* machines, when the SUT is set next boot by efibootmgr, eg to a usb device, the information will NOT be reflected into `ipmitool('chassis bootparam get 5')`, so the ipmitool bootparam output is not up-to-date. But if after efibootmgr controls the boot order, and then use ipmitool to set boot order again, the `ipmitool('chassis bootparam get 5')` is up to date and can control boot order as it expects.
I test on different machines, and the behavior is not exactly the same, but at least above workaround works well. Since we have many different machines from different vendor and with different spec, I expect that this change can make us more guaranteed to boot into pxe, although may sacrifice one time of `ipmitool("chassis bootdev pxe ${options}");`, but worthwhile I think.